grpcclient/grpcclient.go

```diff
@@ -108,9 +115,17 @@ func (cfg *Config) CallOptions() []grpc.CallOption {
 	return opts
 }
 
-// DialOption returns the config as a grpc.DialOptions. The passed inceptors
-// wrap around the configured middleware.
-func (cfg *Config) DialOption(unaryClientInterceptors []grpc.UnaryClientInterceptor, streamClientInterceptors []grpc.StreamClientInterceptor) ([]grpc.DialOption, error) {
+// DialOption returns the config as a grpc.DialOptions. The passed interceptors wrap around the configured middleware.
+// invalidClusterValidation and logger are required to be non-nil, and they are used for reporting the results of the
+// cluster validation, when it is enabled.
+func (cfg *Config) DialOption(unaryClientInterceptors []grpc.UnaryClientInterceptor, streamClientInterceptors []grpc.StreamClientInterceptor, invalidClusterValidations *prometheus.CounterVec, logger log.Logger) ([]grpc.DialOption, error) {
+	if invalidClusterValidations == nil {
+		return nil, fmt.Errorf("invalidClusterValidations must not be null")
+	}
+	if logger == nil {
+		return nil, fmt.Errorf("logger must not be null")
```

Comment:
Being nitpicky, but we don't have nulls in go.

```suggestion
		return nil, fmt.Errorf("invalidClusterValidations must not be nil")
	}
	if logger == nil {
		return nil, fmt.Errorf("logger must not be nil")
```